tasks/ci.rake

```diff
@@ -0,0 +1,282 @@
+        env_config
+      else
+        # By default we assume TEST_TARGET is an agent-only string
+        if agent_target = ENV['TEST_TARGET']
+          master_target = ENV['MASTER_TEST_TARGET'] || DEFAULT_MASTER_TEST_TARGET
+          "#{master_target}-#{agent_target}"
+        else
+          DEFAULT_TEST_TARGETS
+        end
+      end
+
+    if File.exists?(hosts)
+      ENV['HOSTS'] = hosts
+    else
+      hosts_file = "tmp/#{hosts}-#{SecureRandom.uuid}.yaml"
+      cli = BeakerHostGenerator::CLI.new([hosts, '--disable-default-role', '--osinfo-version', '1'])
+      FileUtils.mkdir_p('tmp') # -p ignores when dir already exists
+      File.open(hosts_file, 'w') do |fh|
+        fh.print(cli.execute)
+      end
+      ENV['HOSTS'] = hosts_file
+    end
+  end
+
+  namespace :test do
+    desc <<-EOS
+Run a limited but representative subset of acceptance tests against puppet-agent
+(AIO) packages. This task is intended to reduce testing time on a per-commit
+basis.
+
+  $ SHA=<full sha> bundle exec rake ci:test:quick
+
+SHA should be the full SHA for the puppet-agent package.
+EOS
+    task :quick => ['ci:check_env', 'ci:gen_hosts'] do
+      ENV['TESTS'] = get_test_sample.join(",")
+      Rake::Task["ci:test:aio"].invoke
+    end
+
+    desc <<-EOS
+Run the acceptance tests using puppet-agent (AIO) packages.
+
+  $ SHA=<full sha> bundle exec rake ci:test:aio
+
+SHA should be the full SHA for the puppet-agent package.
+EOS
+    task :aio => ['ci:check_env', 'ci:gen_hosts'] do
+      beaker_suite(:aio)
+    end
+
+    desc <<-EOS
+Run the acceptance tests against puppet gem on various platforms, performing a
+basic smoke test.
+
+  $ SHA=<full sha> bundle exec rake:ci:gem
+
+SHA should be the full SHA for the component.
+EOS
+    task :gem => ['ci:check_env'] do
+      beaker(:init, '--hosts', 'config/nodes/gem.yaml', '--options-file', 'config/gem/options.rb')
+      beaker(:provision)
+      beaker(:exec, 'pre-suite', '--pre-suite', pre_suites(:gem))
+      beaker(:destroy)
+    end
+
+    desc <<-EOS
+Run the acceptance tests against a git checkout.
+
+  $ SHA=<full sha> bundle exec rake ci:test:git
+
+SHA should be the full SHA for the component. Other options:
+
+FORK: to test against your fork, defaults to 'puppetlabs'
+
+SERVER: to git fetch from an alternate GIT server, defaults to 'github.com'
+EOS
+    task :git => ['ci:check_env', 'ci:gen_hosts'] do
+      beaker_suite(:git)
+    end
+  end
+
+  task :test_and_preserve_hosts => ['ci:check_env', 'ci:gen_hosts'] do
+    puts "WARNING, the test_and_preserve_hosts task is deprecated, use ci:test:aio instead."
+    Rake::Task['ci:test:aio'].execute
+  end
+end
+
+task :default do
+  sh('rake -T')
+end
+
+task :spec do
+  sh('rspec lib')
+end
+
+def beaker(command, *argv)
+  argv.concat(ENV['OPTIONS'].split(' ')) if ENV['OPTIONS']
+
+  sh('beaker', command.to_s, *argv)
+end
+
+def beaker_suite(type)
```

Comment:
@joshcooper this is what I have so far. Not necessarily a final version, but at least it's proof this is working!